javalin/src/main/java/io/javalin/http/staticfiles/JettyResourceHandler.kt

```diff
@@ -52,6 +59,49 @@ class JettyResourceHandler(val precompressStaticFiles: Boolean = false) : io.jav
         }
     }
 
+    inner class AliasResourceHandler(private val urlPathPrefix: String, private val aliasCheck: AliasCheck) : ResourceHandler() {
+        override fun getResource(path: String): Resource {
+            val targetResource by lazy { path.removePrefix(urlPathPrefix) }
+            return when {
+                urlPathPrefix == "/" -> getResourceWithAliasChecks(path)!!
+                targetResource == "" -> getResourceWithAliasChecks("/")!! // directory without trailing '/'
+                !path.startsWith(urlPathPrefix) -> EmptyResource.INSTANCE
+                !targetResource.startsWith("/") -> EmptyResource.INSTANCE
+                else -> getResourceWithAliasChecks(targetResource)!!
+            }
+        }
+
+        // same as regular ResourceHandler but with alias checks
+        private fun getResourceWithAliasChecks(path: String) : Resource? {
+            try {
+                var resource: Resource? = null
+                if (baseResource != null) {
+                    val caconicalPath = URIUtil.canonicalPath(path)
+                    resource = baseResource.addPath(caconicalPath)
+                    if (resource != null && resource.isAlias && !checkAlias(path, resource)) {
```

Comment:
This checks `isAlias`, but `isAlias` is also checked in the `checkAlias` method below.

Reply:
I may keep only this one and remove additional from `checkAlias` (it will be more clear)